spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -463,4 +475,11 @@ private void resolveChildEntity(Object entity) {
 											childPersistentEntity)));
 				});
 	}
+
+	private Function<List, Collection<Mutation>> getListCollectionFunction(
```

Comment:
Maybe it's just me, but this a bit confusing. Why don't we have a simple function that takes a list of objects and returns an `Iterable<Mutation>`? I think it all goes back to `applyMutationTwoArgs` which expects a function that generates the mutations rather than the straight forward list of mutations. Why was the `applyMutationTwoArgs` necessary instead of just `writeMutations(Iterable<Mutation>)`?

Reply:
The reason is that there are 4 different types of mutation-generating functions, (insert, update, upsert, delete) and those mutation-generating functions are the variables when refactored into the way it currently is. 

The essence of this PRIVATE function is just the flatMap: each item to write individually is also a collection of mutations because each individual item can result in many mutations of children.